pkg/errors/errors.go

```diff
@@ -0,0 +1,86 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// Common timer domain errors. Use `ErrX.WithCausef()` to clone and add context.
+var (
+	ErrInvalid     = Error{Code: "bad_request", Message: "Request is not valid"}
+	ErrNotFound    = Error{Code: "not_found", Message: "Requested entity not found"}
+	ErrConflict    = Error{Code: "conflict", Message: "An entity with conflicting identifier exists"}
+	ErrInternal    = Error{Code: "internal_error", Message: "Some unexpected error occurred"}
+	ErrUnsupported = Error{Code: "unsupported", Message: "Requested feature is not supported"}
+)
+
+// E converts any given error to the Error type. Unknown are converted
+// to ErrInternal.
+func E(err error) Error {
+	if e, ok := err.(Error); ok {
+		return e
+	}
+	return ErrInternal.WithCausef(err.Error())
+}
+
+// Error represents any error returned by the Entropy components along with any
+// relevant context.
+type Error struct {
+	Code    string `json:"code"`
+	Cause   string `json:"cause,omitempty"`
+	Message string `json:"message"`
+}
+
+// WithCausef returns clone of err with the cause added. Use this when
+// you need to provide description of the underlying technical root-cause
+// which may be written in log for debugging purposes. Cause will be shown
+// to the user only when the Message is empty.
+func (err Error) WithCausef(format string, args ...interface{}) Error {
+	cloned := err
```

Comment:
What's the reason for using another variable here?

Reply:
to make it explicit that it's a copy.